src/components/Notifications/NotificationsLog/NotificationsLogDateFilter.tsx

```diff
@@ -0,0 +1,295 @@
+      const startDate = parseISO(start);
+      setPeriod(
+        produce((draft) => {
+          draft[0] = startDate;
+          if (!draft[1]) {
+            draft[1] = min([add(startDate, { days: 1 }), maxDate]);
+          }
+        })
+      );
+    },
+    [props.setPeriod, maxDate]
+  );
+
+  const setEndDate = React.useCallback(
+    (end: string) => {
+      const setPeriod = props.setPeriod;
+      setPeriod(
+        produce((draft) => {
+          draft[1] = parseISO(end);
+        })
+      );
+    },
+    [props.setPeriod]
+  );
+
+  const startValue: string | undefined = React.useMemo(
+    () => (props.period[0] ? format(props.period[0], 'yyyy-MM-dd') : undefined),
+    [props.period]
+  );
+  const endValue: string | undefined = React.useMemo(
+    () => (props.period[1] ? format(props.period[1], 'yyyy-MM-dd') : undefined),
+    [props.period]
+  );
+
+  const startDateRef = useRef<DatePickerRef>(null);
+  const endDateRef = useRef<DatePickerRef>(null);
+
+  const onClickStartDateInput = React.useCallback(() => {
+    startDateRef.current?.setCalendarOpen(true);
+  }, [startDateRef]);
+
+  const onClickEndDateInput = React.useCallback(() => {
+    endDateRef.current?.setCalendarOpen(true);
+  }, [endDateRef]);
+
+  const startDateInputProps = React.useMemo<TextInputProps>(
+    () => ({
+      ...dateInputProps,
+      onClick: onClickStartDateInput,
+    }),
+    [onClickStartDateInput]
+  );
+
+  const endDateInputProps = React.useMemo<TextInputProps>(
+    () => ({
+      ...dateInputProps,
+      onClick: onClickEndDateInput,
+    }),
+    [onClickEndDateInput]
+  );
+
+  return (
+    <Split>
+      <SplitItem>
+        <DatePicker
+          placeholder="Start"
+          inputProps={startDateInputProps}
+          validators={startRangeValidators}
+          onChange={setStartDate}
+          value={startValue}
+          ref={startDateRef}
+        />
+      </SplitItem>
+      <SplitItem>
+        <Bullseye>
+          <span className={toClassName}>to</span>
+        </Bullseye>
+      </SplitItem>
+      <SplitItem>
+        <DatePicker
+          placeholder="End"
+          inputProps={endDateInputProps}
+          validators={endRangeValidators}
+          onChange={setEndDate}
+          value={endValue}
+          ref={endDateRef}
+        />
+      </SplitItem>
+    </Split>
+  );
+};
+
+export interface NotificationsLogDateFilterProps {
+  value: NotificationsLogDateFilterValue;
+  setValue: (value: NotificationsLogDateFilterValue) => void;
+  retentionDays: number;
+  period: EventPeriod;
+  setPeriod: Dispatch<SetStateAction<EventPeriod>>;
+}
+
+export const NotificationsLogDateFilter: React.FunctionComponent<NotificationsLogDateFilterProps> =
+  (props) => {
+    const options = React.useMemo(
+      () =>
+        Object.values(NotificationsLogDateFilterValue).map((v) => (
+          <SelectOption key={v} value={new EventLogSelectObject(v)} />
+        )),
+      []
+    );
```

Comment:
No need to wrap this in `useMemo` you can directly use the

```
 Object.values(NotificationsLogDateFilterValue).map((v) => (
          <SelectOption key={v} value={new EventLogSelectObject(v)} />
        ))
```

as children of `Select`

Reply:
Fixed